### PR TITLE
{ghi10291} Default Proc Prefab applies correct translation

### PR DESCRIPTION
**Pull request by jackalbe**
{ghi10291} Default Proc Prefab applies correct translation

Signed-off-by: Allen Jackson <23512001+jackalbe@users.noreply.github.com>

## What does this PR do?

* reconfigured the logic so that the MeshData is matched with its own TransformData so that the SetLocalTransform() applies the correct 3x4 Matrix
* A side benefit will make this logic easier to extend in the future to find and apply other metadata about MeshData in the future

### CODE REVIEW

File: Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp [hunk 1]
```diff
@@ -106,31 +106,13 @@ namespace AZ::SceneAPI::Behaviors
         };
 
         using MeshTransformEntry = AZStd::pair<Containers::SceneGraph::NodeIndex, MeshNodeData>;
-        using MeshTransformMap = AZStd::unordered_multimap<Containers::SceneGraph::NodeIndex, MeshNodeData>;
+        using MeshDataMap = AZStd::unordered_multimap<Containers::SceneGraph::NodeIndex, MeshNodeData>; // MeshData Index -> MeshNodeData
         using MeshIndexContainer = AZStd::unordered_set<Containers::SceneGraph::NodeIndex>;
```
Comment (michabr) [inline Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp:110]: MeshIndexContainer no longer needed.

File: Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp [hunk 2]
```diff
@@ -145,49 +127,45 @@ namespace AZ::SceneAPI::Behaviors
             }
 
             MeshIndexContainer meshIndexContainer;
```
Comment (michabr) [inline Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp:129]: meshIndexContainer no longer seems to be needed.

File: Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp [hunk 3]
```diff
@@ -106,31 +106,13 @@ namespace AZ::SceneAPI::Behaviors
         };
 
         using MeshTransformEntry = AZStd::pair<Containers::SceneGraph::NodeIndex, MeshNodeData>;
```
Comment (michabr) [inline Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp:108]: Suggestion: Might be worth renaming MeshTransformEntry to MeshDataEntry.

File: Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp [hunk 4]
```diff
@@ -106,31 +106,13 @@ namespace AZ::SceneAPI::Behaviors
...
-        using MeshTransformMap = AZStd::unordered_multimap<Containers::SceneGraph::NodeIndex, MeshNodeData>;
+        using MeshDataMap = AZStd::unordered_multimap<Containers::SceneGraph::NodeIndex, MeshNodeData>; // MeshData Index -> MeshNodeData
```
Comment (michabr) [inline Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp:109]: Does this still need to be a multimap?

**Review by michabr — APPROVED** [review]
Just have some minor questions.

File: Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp (hunk 3 above)
Comment (jackalbe) [reply Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp:108]: will do!

File: Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp (hunk 4 above)
Comment (jackalbe) [reply Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp:109]: I do no think so, will check this out

**Review by amzn-mike — APPROVED** [review]

**Pull request merged**